### PR TITLE
[CMake] add PYTHON_EXT_SUFFIX back to the main library

### PR DESCRIPTION
**Pull request by nim65s**
Hi,

This PR re-apply https://github.com/stack-of-tasks/eigenpy/pull/57, because the main library is linked to boost python, and in case we have a setup with multiple python versions, we want to ensure that the python module will not try to link to a version of the main library linked to the wrong boost python library.

This is a reccuring issue for users who are using Python 3 on Ubuntu 18.04 or 20.04, while they also have ROS, which is stuck to a Python 2
version of eigenpy.
eg. https://github.com/stack-of-tasks/eigenpy/issues/230

This was removed in https://github.com/stack-of-tasks/eigenpy/pull/113.

Instead of deactivating this feature for everybody because this can raise an issue on Windows (I have no idea what this issue is), this PR reactivate this feature for everybody except Windows users.

@seanyen : as the author of #113, I guess you were trying to fix something, so can you confirm that I am not breaking anything for you here ?

### CODE REVIEW

**Comment by nim65s** [comment]
This was deleted by mistake.

File: CMakeLists.txt [hunk 1]
```diff
@@ -182,6 +182,13 @@ TARGET_INCLUDE_DIRECTORIES(${PROJECT_NAME}
   $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
   $<INSTALL_INTERFACE:include>)
 
+IF(APPLE)
```
Comment (jcarpent) [inline CMakeLists.txt:185]: Why did you change here?

Comment (nim65s) [reply CMakeLists.txt:185]: I'm just reapplying this: https://github.com/stack-of-tasks/eigenpy/commit/8cd9fd9a5d667b267ffbf9918a92237d120fbe49

If this is useless, I can remove it.
